network/endpoint_linux.go

```diff
@@ -144,15 +153,20 @@ func (nw *network) newEndpointImpl(_ apipaClient, nl netlink.NetlinkInterface, p
 				epClient.DeleteEndpointRules(endpt)
 			}
 
-			epClient.DeleteEndpoints(endpt)
+			if nw.extIf != nil {
```

Comment:
What does this condition check ? ( just curious about the variable `extif`)

Reply:
extif contains host interface info and endpoints created with it. this is pointer and if nil, i see test case crashing. fixing it as part of that